src/Middleware/HttpLogging/src/W3CLoggingMiddleware.cs

```diff
@@ -215,6 +217,11 @@ public async Task Invoke(HttpContext context)
                 shouldLog |= AddToList(elements, _protocolStatusIndex, response.StatusCode.ToString(CultureInfo.InvariantCulture));
             }
 
+            if (options.LoggingFields.HasFlag(W3CLoggingFields.TimeTaken))
+            {
+                shouldLog |= AddToList(elements, _timeTakenIndex, stopWatch.GetElapsedTime().TotalMilliseconds.ToString(CultureInfo.InvariantCulture));
```

Comment:
Is the fractional part needed?
```suggestion
                shouldLog |= AddToList(elements, _timeTakenIndex, stopWatch.GetElapsedTime().TotalMilliseconds.ToString("F0", CultureInfo.InvariantCulture));
```